### PR TITLE
Various shell and static analysis improvements

### PR DESCRIPTION
**Pull request by superm1**
I was glancing at codacity and fixed a few things that stood out, and also upped shellcheck to warning.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/redfish/tests/redfish.py [hunk 1]
```diff
@@ -18,9 +18,6 @@
 
 
 def _failure(msg: str, status=400):
-    res = {
-        "error": {"message": msg},
-    }
     return Response(response=json.dumps(msg), status=401, mimetype="application/json")
```
Comment (hughsie) [inline plugins/redfish/tests/redfish.py:21]: I actually think this should have been json.dumps(res) -- does that look plausible?

File: plugins/redfish/tests/redfish.py [hunk 2]
```diff
@@ -97,7 +94,6 @@ def firmware_inventory_bmc():
         "Manufacturer": "Lenovo",
         "Name": "Lenovo BMC Firmware",
         "RelatedItem": [{"@odata.id": "/redfish/v1/Managers/BMC"}],
-        "ReleaseDate": "2017-08-22T12:00:00",
```
Comment (hughsie) [inline plugins/redfish/tests/redfish.py:100]: What was the problem with this?

**Review by hughsie — APPROVED** [review]
two questions, but looks generally okay. Thanks.

File: plugins/redfish/tests/redfish.py (hunk 2 above)
Comment (superm1) [reply plugins/redfish/tests/redfish.py:100]: duplicate `ReleaseDate` key

File: plugins/redfish/tests/redfish.py (hunk 1 above)
Comment (superm1) [reply plugins/redfish/tests/redfish.py:21]: Yeah, looks more reasonable to me.  Will switch.
```
>>> from flask import Response
>>> import json
>>> msg='foo'
>>> Response(response=json.dumps(msg), status=401, mimetype="application/json")
<Response 5 bytes [401 UNAUTHORIZED]>
>>> res = {
...         "error": {"message": msg},
...     }
>>> Response(response=json.dumps(res), status=401, mimetype="application/json")
<Response 29 bytes [401 UNAUTHORIZED]>
```

**Pull request merged**